From: Andreas Gruenbacher <agruen@suse.de>
Subject: Disable file capabilities by default; file_caps=<0|1> option

Disable file capabilities by default: we are still lacking documentation
and file capability awareness in system management tools.

SLES 11 supports a file_caps=<0|1> switch, while the corresponding mailine
switches are called file_caps and no_file_caps.  This patch adds support
for the SLES 11 syntax for compatibility.
(Support for the SLES 11 specific syntax will be removed at some future time.)

Signed-off-by: Andreas Gruenbacher <agruen@suse.de>

---
 Documentation/kernel-parameters.txt |    8 +++++++-
 kernel/capability.c                 |    9 ++++++++-
 2 files changed, 15 insertions(+), 2 deletions(-)

Index: b/Documentation/kernel-parameters.txt
===================================================================
--- a/Documentation/kernel-parameters.txt
+++ b/Documentation/kernel-parameters.txt
@@ -1666,7 +1666,18 @@ and is between 256 and 4096 characters.
 
 	no_file_caps	Tells the kernel not to honor file capabilities.  The
 			only way then for a file to be executed with privilege
-			is to be setuid root or executed by root.
+			is to be setuid root or executed by root. They
+			default to disabled.
+
+	file_caps	Tells the kernel to honor file capabilities.  The
+			only way then for a file to be executed with privilege
+			is to be setuid root or executed by root. They default
+			to disabled.
+
+	file_caps=	Format=<0|1>
+			Same as file_caps and no_file_caps, depending on the value.
+			This variant exists because of backwards compatibility with
+			SLES 11, and will go away at some point in the future.
 
 	nohalt		[IA-64] Tells the kernel not to use the power saving
 			function PAL_HALT_LIGHT when idle. This increases
Index: b/kernel/capability.c
===================================================================
--- a/kernel/capability.c
+++ b/kernel/capability.c
@@ -30,7 +30,7 @@ EXPORT_SYMBOL(__cap_full_set);
 EXPORT_SYMBOL(__cap_init_eff_set);
 
 #ifdef CONFIG_SECURITY_FILE_CAPABILITIES
-int file_caps_enabled = 1;
+int file_caps_enabled;
 
 static int __init file_caps_disable(char *str)
 {
@@ -38,6 +38,13 @@ static int __init file_caps_disable(char
 	return 1;
 }
 __setup("no_file_caps", file_caps_disable);
+
+static int __init file_caps_enable(char *str)
+{
+	file_caps_enabled = str[0] == '\0' || (str[0] == '=' && str[1] == '1');
+	return 1;
+}
+__setup("file_caps", file_caps_enable);
 #endif
 
 /*
